### PR TITLE
CORE-8034 Renamed 'application/rpc-worker' module to 'application/rest-worker', updated helm charts accordingly

### PR DESCRIPTION
**Pull request by ben-millar**
### Description:
This PR satisfies [CORE-8034](https://r3-cev.atlassian.net/browse/CORE-8034); part of the greater renaming initiative tracked by [CORE-7044](https://r3-cev.atlassian.net/browse/CORE-7044).  

[CORE-8034]: https://r3-cev.atlassian.net/browse/CORE-8034?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: applications/workers/release/combined-worker/src/main/kotlin/net/corda/applications/workers/combined/CombinedWorker.kt [hunk 1]
```diff
@@ -181,7 +181,7 @@ private class CombinedWorkerParams {
     @Option(names = ["-d", "--database-params"], description = ["Database parameters for the worker."])
     var databaseParams = emptyMap<String, String>()
 
-    @Option(names = ["-r", "--rpc-params"], description = ["RPC parameters for the worker."])
+    @Option(names = ["-r", "--rest-params"], description = ["RPC parameters for the worker."])
```
Comment (ben-millar) [inline applications/workers/release/combined-worker/src/main/kotlin/net/corda/applications/workers/combined/CombinedWorker.kt]: This description should be changed to 'rest'

File: applications/workers/workers-smoketest/src/smokeTest/kotlin/net/corda/applications/workers/smoketest/virtualnode/VirtualNodeRestTest.kt [hunk 2]
```diff
@@ -38,7 +38,7 @@ import net.corda.e2etest.utilities.getFlowStatus
  */
 @Order(10)
 @TestMethodOrder(MethodOrderer.OrderAnnotation::class)
-class VirtualNodeRpcTest {
+class VirtualNodeRestTest {
```
Comment (ben-millar) [inline applications/workers/workers-smoketest/src/smokeTest/kotlin/net/corda/applications/workers/smoketest/virtualnode/VirtualNodeRestTest.kt]: This change is out of scope and should be reverted.

**Comment by ben-millar** [comment]
This should be unblocked by https://github.com/corda/corda-shared-build-pipeline-steps/pull/566/files

**Comment by ben-millar** [comment]
Merge alongside:
API Changes: https://github.com/corda/corda-api/pull/862
Pipeline Changes: https://github.com/corda/corda-shared-build-pipeline-steps/pull/573

File: .ci/e2eTests/Jenkinsfile [hunk 3]
```diff
@@ -1,4 +1,4 @@
-@Library('corda-shared-build-pipeline-steps@5.0') _
+@Library('corda-shared-build-pipeline-steps@bm/CORE-7040/CORE-8034') _
```
Comment (wzur-r3) [inline .ci/e2eTests/Jenkinsfile]: I know that is for testing, please change it back to `5.0` before merge

File: Jenkinsfile [hunk 4]
```diff
@@ -1,4 +1,4 @@
-@Library('corda-shared-build-pipeline-steps@5.0') _
+@Library('corda-shared-build-pipeline-steps@bm/CORE-7040/CORE-8034') _
```
Comment (wzur-r3) [inline Jenkinsfile]: I know that is for testing, please change it back to 5.0 before merge

**Review by wzur-r3 — CHANGES_REQUESTED** [review]

File: Jenkinsfile (hunk 4 above)
Comment (ben-millar) [reply Jenkinsfile]: For sure. There are some breaking pipeline changes which need to be made in-sync with this, so I'm chatting with @ronanbrowne as to what this should be – possibly >5.0.

File: .ci/e2eTests/Jenkinsfile (hunk 3 above)
Comment (mnesbit) [reply .ci/e2eTests/Jenkinsfile]: Yes definitely don't forget this

**Review by mnesbit — DISMISSED** [review]

**Review by davidcurrie — COMMENTED** [review]
LGTM - will approve once ready to merge

File: .ci/e2eTests/Jenkinsfile [hunk 5]
```diff
@@ -1,4 +1,4 @@
-@Library('corda-shared-build-pipeline-steps@5.0') _
+@Library('corda-shared-build-pipeline-steps@5.0.1') _
```
Comment (ben-millar) [inline .ci/e2eTests/Jenkinsfile:1]: @wzur-r3 Updated version here in line with this PR https://github.com/corda/corda-shared-build-pipeline-steps/pull/573

File: Jenkinsfile [hunk 6]
```diff
@@ -1,4 +1,4 @@
-@Library('corda-shared-build-pipeline-steps@5.0') _
+@Library('corda-shared-build-pipeline-steps@5.0.1') _
```
Comment (ben-millar) [inline Jenkinsfile:1]: @wzur-r3 Updated version here also in line with this PR https://github.com/corda/corda-shared-build-pipeline-steps/pull/573

**Review by ronanbrowne — APPROVED** [review]
LGTM - build

**Review by vkolomeyko — APPROVED** [review]
LGTM

**Review by wzur-r3 — APPROVED** [review]
LGTM for Jenkinsfiles and Gradle files

**Review by davidcurrie — APPROVED** [review]
🚀

**Pull request merged**